server/src/main/java/io/deephaven/server/table/TableServiceGrpcImpl.java

```diff
@@ -276,11 +248,115 @@ public void updateBy(UpdateByRequest request, StreamObserver<ExportedTableCreati
+            }
+            if (dataType == float.class) {
+                return doubleValue.floatValue();
+            }
+            if (dataType == double.class) {
+                return doubleValue;
+            }
+            if (dataType == BigInteger.class) {
+                return BigDecimal.valueOf(doubleValue).toBigInteger();
+            }
+            if (dataType == BigDecimal.class) {
+                return BigDecimal.valueOf(doubleValue);
+            }
+        } else if (literal.hasBoolValue()) {
+            return literal.getBoolValue();
+        }
+        throw new UnsupportedOperationException("Invalid column type for numeric seek: " + dataType);
+    }
+
     @Override
     public void whereIn(WhereInRequest request, StreamObserver<ExportedTableCreationResponse> responseObserver) {
         oneShotOperationWrapper(BatchTableRequest.Operation.OpCase.WHERE_IN, request, responseObserver);
     }
 
+    @Override
+    public void seekRow(SeekRowRequest request, StreamObserver<SeekRowResponse> responseObserver) {
+        GrpcUtil.rpcWrapper(log, responseObserver, () -> {
+            final SessionState session = sessionService.getCurrentSession();
+            final Ticket sourceId = request.getSourceId();
+            if (sourceId.getTicket().isEmpty()) {
+                throw GrpcUtil.statusRuntimeException(Code.FAILED_PRECONDITION, "No consoleId supplied");
+            }
+
+            SessionState.ExportObject<Table> exportedTable =
+                    ticketRouter.resolve(session, sourceId, "sourceId");
+            session.nonExport()
+                    .require(exportedTable)
+                    .onError(responseObserver)
+                    .submit(() -> {
+                        try {
+                            final Table table = exportedTable.get();
+                            final String columnName = request.getColumnName();
+                            final Class<?> dataType = table.getDefinition().getColumn(columnName).getDataType();
+                            final Object seekValue = getSeekValue(request.getSeekValue(), dataType);
+                            final Long result = table.apply(new SeekRow(
+                                    request.getStartingRow(),
+                                    columnName,
+                                    seekValue,
+                                    request.getInsensitive(),
+                                    request.getContains(),
+                                    request.getIsBackward()));
+                            safelyExecuteLocked(responseObserver, () -> {
+                                SeekRowResponse.Builder rowResponse = SeekRowResponse.newBuilder();
+                                responseObserver.onNext(rowResponse.setResultRow(result).build());
+                                responseObserver.onCompleted();
+                            });
+                        } catch (Exception e) {
```

Comment:
shouldn't need a try/catch, any exception that is thrown will already be passed to the onError(responseObserver) above

Reply:
Neat